src/app/common/hocs/withModal.jsx

```diff
@@ -0,0 +1,31 @@
+import { Component } from 'react'
+
+export default function withModal(ChildComponent) {
+  return class MadalWrapper extends Component {
+    constructor(props) {
+      super(props)
+      this.state = {
```

Comment:
maybe better to use hooks in this place. useState + useEffect for a callback fn.

Reply:
because of lifecicle hooks will be redundant in this place